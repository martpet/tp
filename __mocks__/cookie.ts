export default {
  serialize: vi.fn().mockReturnValue('dummyCookie'),
  parse: vi.fn(),
};
