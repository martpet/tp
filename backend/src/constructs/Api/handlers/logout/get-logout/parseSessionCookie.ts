import cookie from 'cookie';

import { cookieName } from '../../../apiUtils';

export const parseSessionCookie = (cookies?: string[]) => {
  try {
    const parsedCookies = cookie.parse(cookies!.join(';'));
    return parsedCookies[cookieName('session')];
  } catch (e) {
    console.error(e);
    return undefined;
  }
};
