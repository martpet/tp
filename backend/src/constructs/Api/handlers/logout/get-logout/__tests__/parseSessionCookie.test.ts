import cookie from 'cookie';

import { parseSessionCookie } from '../parseSessionCookie';

vi.mock('cookie');

const args = [['dummyCookie1', 'dummyCookie2']];

beforeEach(() => {
  vi.mocked(cookie.parse).mockReturnValue({ session: 'dummySessionId' });
});

describe('parseSessionCookie', () => {
  it('calls "cookie.parse" with correct args', () => {
    parseSessionCookie(...args);
    expect(vi.mocked(cookie.parse).mock.calls).toMatchSnapshot();
  });

  it('returns a correct value', () => {
    expect(parseSessionCookie(...args)).toMatchSnapshot();
  });

  describe('when "cookie.parse" throws', () => {
    beforeEach(() => {
      vi.mocked(cookie.parse).mockImplementation(() => {
        throw new Error();
      });
    });

    it('returns a correct value', () => {
      expect(parseSessionCookie(...args)).toMatchSnapshot();
    });
  });

  describe('when "session" cookie is missing', () => {
    beforeEach(() => {
      vi.mocked(cookie.parse).mockReturnValue({});
    });

    it('returns a correct value', () => {
      expect(parseSessionCookie(...args)).toMatchSnapshot();
    });
  });
});
