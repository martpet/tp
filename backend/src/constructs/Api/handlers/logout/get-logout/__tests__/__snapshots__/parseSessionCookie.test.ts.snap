// Vitest Snapshot v1

exports[`parseSessionCookie > calls "cookie.parse" with correct args 1`] = `
[
  [
    "dummyCookie1;dummyCookie2",
  ],
]
`;

exports[`parseSessionCookie > returns a correct value 1`] = `"dummySessionId"`;

exports[`parseSessionCookie > when "cookie.parse" throws > returns a correct value 1`] = `undefined`;

exports[`parseSessionCookie > when "session" cookie is missing > returns a correct value 1`] = `undefined`;
